subprojects/diagnostics/src/main/java/org/gradle/api/tasks/diagnostics/DependencyInsightReportTask.java

```diff
@@ -367,18 +415,62 @@ private void printSection(StyledTextOutput out, Section extraDetail, int depth)
 
         private void printVariantDetails(StyledTextOutput out, RenderableDependency dependency) {
-            List<ResolvedVariantResult> resolvedVariants = dependency.getResolvedVariants();
-            for (ResolvedVariantResult resolvedVariant : resolvedVariants) {
+            if (dependency.getResolvedVariants().isEmpty() && dependency.getAllVariants().isEmpty()) {
+                return;
+            }
+            Set<String> selectedVariantNames = dependency.getResolvedVariants()
+                .stream()
+                .map(ResolvedVariantResult::getDisplayName)
+                .collect(Collectors.toSet());
+            if (task.getShowingAllVariants().get()) {
+                out.style(Header);
                 out.println();
-                out.withStyle(Description).text("   variant \"" + resolvedVariant.getDisplayName() + "\"");
-                AttributeContainer attributes = resolvedVariant.getAttributes();
-                AttributeContainer requested = getRequestedAttributes(configuration, dependency);
-                if (!attributes.isEmpty() || !requested.isEmpty()) {
-                    writeAttributeBlock(out, attributes, requested);
+                out.text("-------------------").println();
+                out.text("Selected Variant(s)").println();
+                out.text("-------------------");
+                out.style(Normal);
+            }
+            for (ResolvedVariantResult variant : dependency.getResolvedVariants()) {
+                printVariant(out, dependency, variant, true);
+            }
+            if (task.getShowingAllVariants().get()) {
+                out.style(Header);
+                out.text("---------------------").println();
+                out.text("Unselected Variant(s)").println();
+                out.text("---------------------");
+                out.style(Normal);
+                for (ResolvedVariantResult variant : dependency.getAllVariants()) {
+                    if (selectedVariantNames.contains(variant.getDisplayName())) {
+                        continue;
+                    }
+                    printVariant(out, dependency, variant, false);
                 }
             }
         }
 
+        private void printVariant(
+            StyledTextOutput out, RenderableDependency dependency, ResolvedVariantResult variant, boolean selected
+        ) {
+            AttributeContainer attributes = variant.getAttributes();
+            AttributeContainer requested = getRequestedAttributes(configuration, dependency);
+            AttributeBuckets buckets = bucketAttributes(attributes, requested);
+
+            out.println().style(Normal).text("Variant ");
+
+            // Style the name based on whether it is selected or not.
+            if (selected) {
+                out.style(Success);
+            } else if (buckets.bothAttributes.values().stream().noneMatch(v -> v.matchType() == MatchType.INCOMPATIBLE)) {
+                out.style(AlternativeSuccess);
```

Comment:
With project dependencies, I see multiple variants that are green. I'm guessing some of them are this alternative success?

Could we just use Info here, which would match the coloring on the attributes:
- Green = compatible + exact match
- Yellow = compatible
- Red = incompatible